File: clients/rust/src/models/create_stream_out.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[non_exhaustive]
#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct CreateStreamOut {
    pub created_at: jiff::Timestamp,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub max_byte_size: Option<u64>,

    pub name: String,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub retention_period_seconds: Option<u64>,

    pub updated_at: jiff::Timestamp,
}

impl CreateStreamOut {
    pub fn new(created_at: jiff::Timestamp, name: String, updated_at: jiff::Timestamp) -> Self {
        Self {
            created_at,
            max_byte_size: None,
            name,
            retention_period_seconds: None,
            updated_at,
        }
    }
}

File: clients/javascript/src/models/createStreamOut.ts
// this file is @generated





export interface CreateStreamOut {
    createdAt: Date;
maxByteSize?: number | null;
name: string;
retentionPeriodSeconds?: number | null;
updatedAt: Date;
}

export const CreateStreamOutSerializer = {
    _fromJsonObject(object: any): CreateStreamOut {
        return {
            createdAt: new Date(object['created_at']),
            maxByteSize: object['max_byte_size'],
            name: object['name'],
            retentionPeriodSeconds: object['retention_period_seconds'],
            updatedAt: new Date(object['updated_at']),
            };
    },

    _toJsonObject(self: CreateStreamOut): any {
        return {
            'created_at': self.createdAt,
            'max_byte_size': self.maxByteSize,
            'name': self.name,
            'retention_period_seconds': self.retentionPeriodSeconds,
            'updated_at': self.updatedAt,
            };
    }
}